solutionbox/code_free_ml/mltoolbox/code_free_ml/data/local_predict.py

```diff
@@ -0,0 +1,251 @@
+
+  Returns:
+    List of dicts in the form {output_alias_friendly_name: predicted_value}.
+  """
+  _, csv_tensor_name = input_alias_map.items()[0]
+  raw_result = session.run(fetches=output_alias_map,
+                           feed_dict={csv_tensor_name: batch_csv_strings})
+
+  # Convert from numpy arrays to python lists
+  result = {key: raw_result[key].flatten().tolist() for key in raw_result}
+
+  batch_size = len(batch_csv_strings)
+  json_list = []
+  for i in range(batch_size):
+    json_result = {key: result[key][i] for key in result}
+    json_list.append(json_result)
+
+  return json_list
+
+
+def format_results(output_format, output_schema, batched_json_results):
+  """Formats prediction results.
+
+  Args:
+    output_format: 'csv' or 'json'
+    output_schema: output file schema. Only used if output_format is csv. This
+        is where the csv column order is defined.
+    batched_json_results: list of dicts. The output of run_batch_prediction().
+
+  Returns:
+    List of csv lines or a list of json strings.
+  """
+  if output_format == 'csv':
+    batched_results = []
+    for one_json_result in batched_json_results:
+      values = [str(one_json_result[schema['name']])
+                for schema in output_schema]
+      batched_results.append(','.join(values))
+  elif output_format == 'json':
+    batched_results = [json.dumps(x) for x in batched_json_results]
+  else:
+    raise ValueError('Unknown output_format %s' % output_format)
+
+  return batched_results
+
+
+def write_schema_file(output_location, output_schema):
+  """Writes the schema file."""
+  file_io.write_string_to_file(
+      os.path.join(output_location, 'schema.json'),
+      json.dumps(output_schema, indent=2))
+
+
+def append_batched_strings(file_path, batched_formatted_strings):
+  """Appends a list of strings to a text file."""
+  with file_io.FileIO(file_path, 'a') as f:
+    for line in batched_formatted_strings:
+      f.write(line + '\n')
+
+
+def get_output_schema(session, output_alias_map):
+  """Makes output schema.
+
+  Args:
+    session: tensorflow session
+    output_alias_map: dict of {friendly_name: tensor_name}. The session's graph
+        is searched for the tensor's datatype.
+
+  Returns:
+    A bigquery-stype schema object.
+  """
+  schema = []
+  for name in sorted(six.iterkeys(output_alias_map)):
+    tensor_name = output_alias_map[name]
+    dtype = session.graph.get_tensor_by_name(tensor_name).dtype
+    if dtype == tf.int32 or dtype == tf.int64:
+      schema.append({'name': name, 'type': 'INTEGER'})
+    elif dtype == tf.float32 or dtype == tf.float64:
+      schema.append({'name': name, 'type': 'FLOAT'})
+    else:
+      schema.append({'name': name, 'type': 'STRING'})
+
+  return schema
+
+
+def main(argv=None):
+  """Runs local batch prediction."""
+  args = parse_arguments(sys.argv if argv is None else argv)
+
+  file_io.recursive_create_dir(args.output_location)
+  with tf.Graph().as_default(), tf.Session() as sess:
```

Comment:
I feel "run_batch_prediction" can include the logic of loading model and run session so all TF stuff is wrapped into one function. What do you think?